File: typescript/rna-transcription/rna-transcription.ts
interface dnaToRna {
  [dnaNucleotide: string]: string
}

const conversions: dnaToRna = {
  C: 'G',
  G: 'C',
  A: 'U',
  T: 'A'
}

class Transcriptor {
  toRna(dnaStrand: string) {
    return dnaStrand.split('').map(
      (nucleotide: string) => this.convertNucleotide(nucleotide)
    ).join('');
  }
  convertNucleotide(dnaNucleotide: string): string {
    if (!conversions[dnaNucleotide]) {
      throw new Error("Invalid input DNA.");
    }
    return conversions[dnaNucleotide];
  }
}

export default Transcriptor

File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    rna_strand = ''
    for nucleotide in dna_strand:
        rna_strand = rna_strand + convert(nucleotide)
    return rna_strand

def convert(dna_nucleotide):
    if dna_nucleotide == 'G':
        return 'C'
    elif dna_nucleotide == 'C':
        return 'G'
    elif dna_nucleotide == 'T':
        return 'A'
    elif dna_nucleotide == 'A':
        return 'U'
    else:
        raise Exception("Invalid input")
